### PR TITLE
Static branch fixups

### PR DESCRIPTION
**Pull request by billschereriii**
Assorted changes in support of static builds:

- added pytext fixture for link mode
- static and shared builds are now done in separate trees since CMake doesn't seem to consider them as distinct configurations
- Test files updated to build and run with the new paths
- Examples update to build with the new path and run with CWD=source file location directory
- CI/CD updated to perform static builds of examples on non-MacOS platforms

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -152,65 +152,70 @@ test-deps-gpu: test-deps
 
 # help: build-tests                    - build all tests (C, C++, Fortran)
 .PHONY: build-tests
+build-tests: test-deps
```
Comment (ashao) [inline Makefile:155]: I think I'm actually walking back this suggestion that I had made. I think building the tests are useful enough (without running them) that maybe we just add the dependency of the CPP tests on `catch` to this list?

File: examples/parallel/fortran/CMakeLists.txt [hunk 2]
```diff
@@ -42,33 +43,75 @@ if(NOT DEFINED SMARTREDIS_INSTALL_PATH)
 endif()
 
 # Locate dependencies
+#  . MPI
 find_package(MPI REQUIRED)
 IF(NOT MPI_Fortran_FOUND)
 	message(FATAL_ERROR "Could not find Fortran MPI components")
 endif()
+#  . Main SmartRedis Library (C++ based)
```
Comment (ashao) [inline examples/parallel/fortran/CMakeLists.txt:51]: More accurate to say C/C++

File: examples/parallel/fortran/CMakeLists.txt [hunk 3]
```diff
@@ -42,33 +43,75 @@ if(NOT DEFINED SMARTREDIS_INSTALL_PATH)
...
+add_library(smartredis-main ${SMARTREDIS_LINK_MODE} IMPORTED)
 find_library(SR_LIB ${SMARTREDIS_LIB}
     PATHS ${SMARTREDIS_INSTALL_PATH}/lib NO_DEFAULT_PATH
     REQUIRED
     ${SMARTREDIS_LINK_MODE}
 )
+set_target_properties(smartredis-main PROPERTIES
+    IMPORTED_LOCATION ${SR_LIB}
+)
+#  . SmartRedis Fortran Library (C++ based)
```
Comment (ashao) [inline examples/parallel/fortran/CMakeLists.txt:61]: `(C++ based)` is not innacurate

File: .github/workflows/run_tests.yml [hunk 4]
```diff
@@ -145,6 +145,18 @@ jobs:
                 TF /usr/lib/redis/modules/backends/redisai_tensorflow/redisai_tensorflow.so \
                 TORCH /usr/lib/redis/modules/backends/redisai_torch/redisai_torch.so" >> $GITHUB_ENV
 
+      # Check the static build by compiling and linking examples
```
Comment (ashao) [inline .github/workflows/run_tests.yml:148]: While I know that it's "free" to run our tests on Github Actions, what do you think about doing a static build first, then doing a make clean, and then continuing on?

I don't foresee a problem with actually executing the tests once they link correctly.

**Review by ashao — CHANGES_REQUESTED** [review]
Thanks for tracking down all the CMake magic to make this happen. Some very small comments (except potentially the one about test changes).

File: .github/workflows/run_tests.yml (hunk 4 above)
Comment (ashao) [reply .github/workflows/run_tests.yml:148]: Disregard. I see this is what you're already doing

File: examples/parallel/fortran/CMakeLists.txt (hunk 2 above)
Comment (billschereriii) [reply examples/parallel/fortran/CMakeLists.txt:51]: Fixed

File: examples/parallel/fortran/CMakeLists.txt (hunk 3 above)
Comment (billschereriii) [reply examples/parallel/fortran/CMakeLists.txt:61]: Fixed

File: Makefile (hunk 1 above)
Comment (billschereriii) [reply Makefile:155]: Skipping per discussion

File: .github/workflows/run_tests.yml (hunk 4 above)
Comment (billschereriii) [reply .github/workflows/run_tests.yml:148]: static and shared build in separate directories, so the risk of cross-contagion is minimal -- and if it happens, we want to know about it anyway

**Review by ashao — APPROVED** [review]

**Pull request merged**